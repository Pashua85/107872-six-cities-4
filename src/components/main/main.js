import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import CitiesPlaceCardList from '../cities-place-card-list/cities-place-card-list';
import CitiesMap from '../cities-map/cities-map';
import CitiesList from '../cities-list/cities-list';
import PlacesSorting from '../places-sorting/places-sorting';
import EmptyMain from '../empty-main/empty-main';
import {getCurrentCity} from '../../store/reducers/currentCityReducer/selectors';
import {getCityOffers} from '../../store/reducers/offersReducer/selectors';

const Main = (props) => {
  const {places, city} = props;
  const placesAmount = places.length;

  return (
    <div className="page page--gray page--main">
      <header className="header">
        <div className="container">
          <div className="header__wrapper">
            <div className="header__left">
              <a className="header__logo-link header__logo-link--active">
                <img className="header__logo" src="img/logo.svg" alt="6 cities logo" width="81" height="41" />
              </a>
            </div>
            <nav className="header__nav">
              <ul className="header__nav-list">
                <li className="header__nav-item user">
                  <a className="header__nav-link header__nav-link--profile" href="#">
                    <div className="header__avatar-wrapper user__avatar-wrapper">
                    </div>
                    <span className="header__user-name user__name">Oliver.conner@gmail.com</span>
                  </a>
                </li>
              </ul>
            </nav>
          </div>
        </div>
      </header>

      <main className="page__main page__main--index">
        <h1 className="visually-hidden">Cities</h1>
        <div className="tabs">
          <CitiesList />
        </div>
        <div className="cities">
          {
            placesAmount > 0 &&
            (
              <div className="cities__places-container container">
                <section className="cities__places places">
                  <h2 className="visually-hidden">Places</h2>
                  <b className="places__found">{placesAmount} places to stay in {city}</b>
                  <PlacesSorting />
                  <CitiesPlaceCardList />
                </section>
                <div className="cities__right-section">
                  <CitiesMap />
                </div>
              </div>
            )
          }
          {
            placesAmount === 0 &&
            (
              <EmptyMain />
            )
          }
        </div>
      </main>
    </div>
  );
};

Main.propTypes = {
  places: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.string.isRequired,
        propertyName: PropTypes.string.isRequired,
        propertyType: PropTypes.oneOf([`apartment`, `room`, `house`, `hotel`]),
        propertyText: PropTypes.arrayOf(PropTypes.string),
        price: PropTypes.number.isRequired,
        rating: PropTypes.number.isRequired,
        isPremium: PropTypes.bool.isRequired,
        bedroomsAmount: PropTypes.number.isRequired,
        guestMax: PropTypes.number.isRequired,
        propertyItems: PropTypes.arrayOf(PropTypes.string),
        host: PropTypes.shape({
          name: PropTypes.string,
          avatar: PropTypes.string,
          isSuper: PropTypes.bool
        }).isRequired,
        titlePhoto: PropTypes.string.isRequired,
        photos: PropTypes.arrayOf(
            PropTypes.shape({
              title: PropTypes.string,
              id: PropTypes.string
            })
        ),
        reviews: PropTypes.arrayOf(
            PropTypes.shape({
              id: PropTypes.string,
              userName: PropTypes.string,
              avatar: PropTypes.string,
              rating: PropTypes.number,
              text: PropTypes.string
            })
        ),
        coords: PropTypes.arrayOf(PropTypes.number).isRequired
      })
  ),
  city: PropTypes.string.isRequired
};

const mapStateToProps = (state) => ({
  places: getCityOffers(state),
  city: getCurrentCity(state)
});

export default connect(mapStateToProps)(Main);
export {Main};

