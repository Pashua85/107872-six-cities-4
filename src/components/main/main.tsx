import React from 'react';
import {connect} from 'react-redux';
import CitiesPlaceCardList from '../cities-place-card-list/cities-place-card-list';
import CitiesMap from '../cities-map/cities-map';
import CitiesList from '../cities-list/cities-list';
import PlacesSorting from '../places-sorting/places-sorting';
import EmptyMain from '../empty-main/empty-main';
import Header from '../header/header';
import {getCurrentCity} from '../../store/reducers/current-city-reducer/selectors';
import {getCityOffers} from '../../store/reducers/offersReducer/selectors';
import {IPlace, ICity, IStore} from '../../types/types';

interface MainProps {
  places: IPlace[],
  city: ICity
}

const Main: React.FC<MainProps> = (props) => {
  const {places, city} = props;
  const placesAmount = places.length;

  return (
    <div className="page page--gray page--main">

      <Header />

      <main className="page__main page__main--index">
        <h1 className="visually-hidden">Cities</h1>
        <div className="tabs">
          <CitiesList />
        </div>
        <div className="cities">
          {
            placesAmount > 0 &&
            (
              <div className="cities__places-container container">
                <section className="cities__places places">
                  <h2 className="visually-hidden">Places</h2>
                  <b className="places__found">{placesAmount} places to stay in {city.name}</b>
                  <PlacesSorting />
                  <CitiesPlaceCardList />
                </section>
                <div className="cities__right-section">
                  <CitiesMap />
                </div>
              </div>
            )
          }
          {
            placesAmount === 0 &&
            (
              <EmptyMain />
            )
          }
        </div>
      </main>
    </div>
  );
};

const mapStateToProps = (state: IStore) => ({
  places: getCityOffers(state),
  city: getCurrentCity(state)
});

export default connect(mapStateToProps)(Main);
export {Main};

