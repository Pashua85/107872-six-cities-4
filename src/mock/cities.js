import {v4 as uuidv4} from 'uuid';

export default [
  {
    id: uuidv4(),
    cityName: `Paris`,
    places: []
  },
  {
    id: uuidv4(),
    cityName: `Cologne`,
    places: []
  },
  {
    id: uuidv4(),
    cityName: `Brussels`,
    places: []
  },
  {
    id: uuidv4(),
    cityName: `Amsterdam`,
    places: [
      {
        id: uuidv4(),
        propertyName: `Beautiful luxurious apartment at great location`,
        propertyType: `apartment`,
        propertyText: [
          `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
          `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
        ],
        price: 120,
        rating: 3.9,
        isPremium: true,
        bedroomsAmount: 3,
        guestMax: 4,
        propertyItems: [`Wi-Fi`, `Washing machine`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Kitchen`, `Dishwasher`, `Cabel TV`, `Fridge`],
        host: {
          name: `Angelina`,
          avatar: `avatar-angelina.jpg`,
          isSuper: true
        },
        titlePhoto: `apartment-01.jpg`,
        photos: [
          {
            title: `room.jpg`,
            id: uuidv4()
          },
          {
            title: `apartment-01.jpg`,
            id: uuidv4()
          },
          {
            title: `apartment-02.jpg`,
            id: uuidv4()
          },
          {
            title: `apartment-03.jpg`,
            id: uuidv4()
          },
          {
            title: `studio-01.jpg`,
            id: uuidv4()
          },
          {
            title: `apartment-01.jpg`,
            id: uuidv4()
          }
        ],
        reviews: [
          {
            id: uuidv4(),
            userName: `Max`,
            avatar: `img/avatar-max.jpg`,
            rating: 4,
            text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`
          }
        ],
        coords: [52.3909553943508, 4.85309666406198]
      },
      {
        id: uuidv4(),
        propertyName: `Wood and stone place`,
        propertyType: `room`,
        propertyText: [
          `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
          `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
        ],
        price: 80,
        rating: 4.6,
        isPremium: false,
        bedroomsAmount: 1,
        guestMax: 2,
        propertyItems: [`Wi-Fi`, `Washing machine`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Kitchen`, `Dishwasher`, `Cabel TV`, `Fridge`],
        host: {
          name: `Angelina`,
          avatar: `avatar-angelina.jpg`,
          isSuper: true
        },
        titlePhoto: `room.jpg`,
        photos: [
          {
            title: `room.jpg`,
            id: uuidv4()
          },
          {
            title: `apartment-01.jpg`,
            id: uuidv4()
          },
          {
            title: `apartment-02.jpg`,
            id: uuidv4()
          },
          {
            title: `apartment-03.jpg`,
            id: uuidv4()
          },
          {
            title: `studio-01.jpg`,
            id: uuidv4()
          },
          {
            title: `apartment-01.jpg`,
            id: uuidv4()
          }
        ],
        reviews: [],
        coords: [52.369553943508, 4.85309666406198]
      },
      {
        id: uuidv4(),
        propertyName: `Canal View Prinsengracht`,
        propertyType: `apartment`,
        propertyText: [
          `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
          `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
        ],
        price: 132,
        rating: 4.7,
        isPremium: false,
        bedroomsAmount: 3,
        guestMax: 4,
        propertyItems: [`Wi-Fi`, `Washing machine`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Kitchen`, `Dishwasher`, `Cabel TV`, `Fridge`],
        host: {
          name: `Angelina`,
          avatar: `avatar-angelina.jpg`,
          isSuper: true
        },
        titlePhoto: `apartment-02.jpg`,
        photos: [
          {
            title: `room.jpg`,
            id: uuidv4()
          },
          {
            title: `apartment-01.jpg`,
            id: uuidv4()
          },
          {
            title: `apartment-02.jpg`,
            id: uuidv4()
          },
          {
            title: `apartment-03.jpg`,
            id: uuidv4()
          },
          {
            title: `studio-01.jpg`,
            id: uuidv4()
          },
          {
            title: `apartment-01.jpg`,
            id: uuidv4()
          }
        ],
        reviews: [],
        coords: [52.3909553943508, 4.929309666406198]
      },
      {
        id: uuidv4(),
        propertyName: `Nice, cozy, warm big bed apartment`,
        propertyType: `apartment`,
        propertyText: [
          `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
          `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
        ],
        price: 180,
        rating: 5,
        isPremium: true,
        bedroomsAmount: 4,
        guestMax: 5,
        propertyItems: [`Wi-Fi`, `Washing machine`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Kitchen`, `Dishwasher`, `Cabel TV`, `Fridge`],
        host: {
          name: `Angelina`,
          avatar: `avatar-angelina.jpg`,
          isSuper: true
        },
        titlePhoto: `apartment-03.jpg`,
        photos: [
          {
            title: `room.jpg`,
            id: uuidv4()
          },
          {
            title: `apartment-01.jpg`,
            id: uuidv4()
          },
          {
            title: `apartment-02.jpg`,
            id: uuidv4()
          },
          {
            title: `apartment-03.jpg`,
            id: uuidv4()
          },
          {
            title: `studio-01.jpg`,
            id: uuidv4()
          },
          {
            title: `apartment-01.jpg`,
            id: uuidv4()
          }
        ],
        reviews: [],
        coords: [52.3809553943508, 4.939309666406198]
      }
    ]
  },
  {
    id: uuidv4(),
    cityName: `Hamburg`,
    places: []
  },
  {
    id: uuidv4(),
    cityName: `Dusseldorf`,
    places: []
  }
];
