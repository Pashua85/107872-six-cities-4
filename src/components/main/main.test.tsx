import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {Main} from './main';
import places from '../../mock/test-places';
import cities from '../../mock/test-cities';

Enzyme.configure({
  adapter: new Adapter()
});

describe(`Main`, () => {
  it(`should render correctly with four places`, () => {
    const wrapper = shallow(
        <Main places={places} city={cities[1]} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it(`should render correctly with two places`, () => {
    const newPlaces = [places[0], places[2]];
    const wrapper = shallow(
        <Main places={newPlaces} city={cities[0]} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
