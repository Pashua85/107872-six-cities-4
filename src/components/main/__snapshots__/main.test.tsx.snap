// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should render correctly with four places 1`] = `ShallowWrapper {}`;

exports[`Main should render correctly with two places 1`] = `ShallowWrapper {}`;
